src/Compilers/Core/Portable/Diagnostic/LocalizableResourceString.cs

```diff
@@ -113,5 +113,23 @@ public override string ToString(IFormatProvider formatProvider)
                 (_formatArguments.Length > 0 ? string.Format(resourceString, _formatArguments) : resourceString) :
                 string.Empty;
         }
+
+        public override bool Equals(LocalizableString other)
```

Comment:
Yes, they are all localizable string, and we ensure non-null value in descriptor's constructor.

I think you are probably right and == should also do value equality, just like for strings. I think someone from API team needs to decide, I'll tag them in this PR for reviewing public APIs